src/neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -144,6 +149,22 @@ public Witness[] Witnesses
             set { witnesses = value; _size = 0; }
         }
 
+        static Transaction()
+        {
+            using ScriptBuilder sb = new ScriptBuilder();
+            sb.EmitAppCall(NativeContract.Oracle.Hash, "finish");
+            oracleResponseScript = sb.ToArray();
+        }
+
+        private bool CheckOracleResponse(StoreView snapshot)
+        {
+            OracleResponse response = Attributes.OfType<OracleResponse>().FirstOrDefault();
+            if (response is null) return true;
+            OracleRequest request = NativeContract.Oracle.GetRequest(snapshot, response.Id);
+            Transaction request_tx = snapshot.GetTransaction(request.Txid);
+            return Sender.Equals(request_tx.Sender);
```

Comment:
Just checking the `Sender`, @erikzhang?

Reply:
Did I forget something?

Reply:
If it's oracle response, the sender must be oracle nodes, isn't it?

Reply:
The oracle nodes send the transaction, but the `sender` field should be the same as the request transaction. Because it charges the sender.

Reply:
Not support.The actor of replacing the sender is not good, giving the platform too much power.

Reply:
But we have limited the script of the response transaction, it is fixed. So the platform has no power.

https://github.com/neo-project/neo/blob/63b3a759f178a6c30d42844553b85f397c8376ad/src/neo/Network/P2P/Payloads/Transaction.cs#L219-L220

Reply:
> The oracle nodes send the transaction, but the sender field should be the same as the request transaction. Because it charges the sender.

And how the oracle will sign this transaction?